pkg/cloud/services/gc/ec2.go

```diff
@@ -28,6 +28,45 @@ import (
 	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/filter"
 )
 
+func (s *Service) deleteEC2Instances(ctx context.Context, resources []*AWSResource) error {
+	for _, resource := range resources {
+		if !s.isEC2InstanceToDelete(resource) {
+			s.scope.Debug("Resource not an EC2 instance for deletion", "arn", resource.ARN.String())
+			continue
+		}
+
+		instanceID := strings.ReplaceAll(resource.ARN.Resource, "instance/", "")
+		if err := s.deleteEC2Instance(ctx, instanceID); err != nil {
+			return fmt.Errorf("deleting EC2 instance %s: %w", instanceID, err)
```

Comment:
Sorry, I have a new request here because I stumbled over non-actionable error messages in existing GC code (they were missing the region and humans shouldn't be required to loop through all accounts and regions to find an object where only the ID is logged). I'm fixing existing code in a new PR.

```suggestion
			return fmt.Errorf("deleting EC2 instance %s with ID %s: %w", resource.ARN, instanceID, err)
```